batch-setup/make_meta_tiles.py

```diff
@@ -281,33 +357,113 @@ def _big_jobs(rawr_bucket, prefix, key_format_type, rawr_zoom, group_zoom,
         for y in range(num_coords):
             coord = Coordinate(zoom=group_zoom, column=x, row=y)
             tasks.append(p.apply_async(job_sizer, (coord,)))
+            grouping_queue.put_nowait(coord)  # queue for future size counting
 
         # collect tasks and put them into the big jobs list.
         for task in tasks:
-            coord, size = task.get()
-            if size >= size_threshold:
-                big_jobs[coord] = True
+            coord, sizes = task.get()
+            all_sizes.update(sizes)
+
+    # now use all_sizes plus the size_threshold to find the lowest zoom we can group each coordinate into
+    counts_at_zoom = {}
+    while not grouping_queue.empty():
+        this_coord = grouping_queue.get_nowait()
+        this_size = all_sizes[this_coord]
+
+        if this_size <= size_threshold or this_coord.zoom == rawr_zoom:
+            # we're good on size, or can't group at a higher zoom
+            grouped_by_rawr_tile_size.append(this_coord)
+            grouping_queue.task_done()
+
+            if this_coord.zoom not in counts_at_zoom:
+                counts_at_zoom[this_coord.zoom] = 0
+            counts_at_zoom[this_coord.zoom] += 1
+        else:
+            # too big for this zoom, queue the children
+            top_left_child = this_coord.zoomBy(1)
```

Comment:
I thiiink the tile object has a `children()` function that returns this list so you don't have to calculate it?